kmir/src/kmir/kdist/mir-semantics/rt/data.md

```diff
@@ -890,6 +954,25 @@ Error cases for `castKindIntToInt`
     [owise]
 ```
 
+### Pointer Casts
+
+Casting between two raw pointers. FIXME: No validity checks are currently performed
+
+```k
+  // FIXME: Address and Offset are blindly transferred through
+  rule <k> #cast( typedValue(Ptr(DEPTH, PLACE, PTR_MUT, ADDRESS, OFFSET), _TY_FROM, LOCAL_MUT), castKindPtrToPtr, TY_TO)
+          =>
+           typedValue(Ptr(DEPTH, PLACE, PTR_MUT, ADDRESS, OFFSET), TY_TO, LOCAL_MUT)
+          ...
+       </k>
+  //     <types> TYPEMAP </types>
+  //  requires TY_TO #in TYPEMAP
+  //  requires #is_valid_cast(TY_FROM.layout(), TY_TO.layout())
+  
+  rule <k> #cast( typedValue(VALUE, _TY_FROM, LOCAL_MUT), castKindTransmute, TY_TO) => typedValue(VALUE, TY_TO, LOCAL_MUT) ... </k>
```

Comment:
Well... this rule is certainly too generic.
Right now we only want to cast raw pointers and return their "Address" (IIUC, or something that is derived from it?), as a `usize` value that will suit the example program...